### ISSUE TITLE
lateinit property nexusClient has not been initialized

### ISSUE THREAD

**Issue by nicolasfara** [DanySK/publish-on-central#524]
When executing `./gradlew drop` the following error is raised:

```
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':dropStagingRepositoryOnMavenCentral'.
> lateinit property nexusClient has not been initialized
```

**Comment by DanySK** [comment]
Ah! Drop was not designed to be used alone. I need to look into this

**Comment by DanySK** [comment]
Can you provide a build scan?

**Comment by nicolasfara** [comment]
> Can you provide a build scan?

Sure: https://scans.gradle.com/s/bv4q2b3idnzcw

**Comment by DanySK** [comment]
[the culprit](https://github.com/DanySK/publish-on-central/blob/07922b99d235ff6070fdf5f9730680cbcb58de52/src/main/kotlin/org/danilopianini/gradle/mavencentral/InitializeNexusClient.kt#L20)

**Comment by DanySK** [comment]
We need to rethink what the plugin is meant to do, because it is currently designed for a single-run pipeline maven-style.
I want to preserve the pipeline, so something like `uploadAll release` must work (hence, `release` `dependsOn` `close`).
OTOH, `release` should work on a repo that has already been closed.

Nexus related operations:
* `createNexusClient` 
* `createStagingRepository`
* `upload`
* `close`
* `release`
* `drop`

* `createNexusClient` must always run if any of the Nexus-related operations are invoked.
* `createStagingRepository` has the wrong name; it should be `findOrCreateStagingRepository`, the way it is implemented now.
* `upload` tasks are optional. They just need to be ran after `createStagingRepository`
* `close` must run after `upload` and gracefully skip if the repo is already closed
*  `release` must run after `close`
* `drop` is mutually exclusive with `release`, must run after `close`, must run after all the uploads whether or not `close` is scheduled

This should support both a "one shot" pipeline and a multi-stage pipeline, doesn't it?

**Comment by DanySK** [comment]
@nicolasfara I prepared a patch in #525, but I'd like to have a regression test before merging. Can you prepare an additional test to be performed in CI and target branch `test/524` with a PR? I will then open a PR `test/524 -> master` to ensure that the test "fails correctly" and a PR `fix/525 -> test/524` to ensure that it is indeed fixed.

**Comment by DanySK** [comment]
Fixed in #525 which will be merged soon. Reopen if the fix does not work.
